src/Bybit/Bybit.jl

```diff
@@ -0,0 +1,180 @@
+module Bybit
+
+export BybitCommonQuery,
+    BybitPublicQuery,
+    BybitAccessQuery,
+    BybitPrivateQuery,
+    BybitAPIError,
+    BybitClient,
+    BybitData
+
+using Serde
+using Dates, NanoDates, TimeZones, Base64, Nettle
+
+using ..CryptoAPIs
+import ..CryptoAPIs: Maybe, AbstractAPIsError, AbstractAPIsData, AbstractAPIsQuery, AbstractAPIsClient
+
+
+abstract type BybitData <: AbstractAPIsData end
+abstract type BybitCommonQuery  <: AbstractAPIsQuery end
+abstract type BybitPublicQuery  <: BybitCommonQuery end
+abstract type BybitAccessQuery  <: BybitCommonQuery end
+abstract type BybitPrivateQuery <: BybitCommonQuery end
+
+"""
+    List{D} <: AbstractAPIsData
+
+## Required fields
+- `list::Vector{D}`: Result values.
+
+## Optional fields
+- `nextPageCursor::String`: Cursor. Used to pagination.
+- `category::String`: Product type
+"""
+struct List{D<:AbstractAPIsData} <: AbstractAPIsData
+    list::Vector{D}
+    nextPageCursor::Maybe{String}
+    category::Maybe{String}
+end
+
+"""
+    Rows{D} <: AbstractAPIsData
+
+## Required fields
+- `rows::Vector{D}`: Result values.
+
+## Optional fields
+- `nextPageCursor::String`: Cursor. Used to pagination.
+"""
+struct Rows{D<:AbstractAPIsData} <: AbstractAPIsData
+    rows::Vector{D}
+    nextPageCursor::Maybe{String}
+end
+
+"""
+    Data{D} <: AbstractAPIsData
+
+## Required fields
+- `retCode::Int64`: Return code.
+- `retMsg::String`: Return message.
+- `result::D`: Request result.
+- `retExtInfo::Dict{String,Any}`: Request extended information.
+- `time::NanoDate`: Time of the request.
+"""
+struct Data{D<:Maybe{A} where {A<:AbstractAPIsData}} <: AbstractAPIsData
+    retCode::Int64
+    retMsg::String
+    result::D
+    retExtInfo::Dict{String,Any}
+    time::NanoDate
+end
+
+"""
+    BybitClient <: AbstractAPIsClient
+
+Client info.
+
+## Required fields
+- `base_url::String`: Base URL for the client. 
+
+## Optional fields
+- `public_key::String`: Public key for authentication.
+- `secret_key::String`: Secret key for authentication.
+- `interface::String`: Interface for the client.
+- `proxy::String`: Proxy information for the client.
+- `account_name::String`: Account name associated with the client.
+- `description::String`: Description of the client.
+"""
+Base.@kwdef struct BybitClient <: AbstractAPIsClient
+    base_url::String
+    public_key::Maybe{String} = nothing
+    secret_key::Maybe{String} = nothing
+    interface::Maybe{String} = nothing
+    proxy::Maybe{String} = nothing
+    account_name::Maybe{String} = nothing
+    description::Maybe{String} = nothing
+end
+
+"""
+    BybitAPIError{T} <: AbstractAPIsError
+
+Exception thrown when an API method fails with code `T`.
+
+## Required fields
+- `retCode::Int64`: Error code.
+- `retMsg::String`: Error message.
+- `time::NanoDate`: Error time.
+
+## Optional fields
+- `result::String`: Error result.
+- `retExtInfo::String`: Extended error information.
```

Comment:
why are `result` and `retExtInfo` optional fields?

Reply:
and they have type `Dict{String,Any}`

Reply:
if `result` and `retExtInfo` optional fields, then they must have the type `Maybe{Dict{String,Any}}`